### DISCUSSION TITLE
how to appropriately set up analyses depending on nested corpora (that cannot be shared) in terms of access & dependencies?

### DISCUSSION THREAD

**Discussion by alecristia** [LAAC-LSCP/ChildProject#302] [Q&A] [answered]
By now, we've converged on the following system for paper write-up, with two levels of nesting:

- the top level is the paper
- one level down in nesting, there is a metrics repo
- another level down, that metrics repo has nested in it one or more corpora, which contain `metadata/children.csv` and `metadata/recordings.csv`

We should always be able to share the top level, such that anyone, even people who do not have access to the corpora, can reproduce it. But right now, that's not the case: My paper write-ups always read `metadata/children.csv` and `metadata/recordings.csv`. See [sample issue](https://github.com/aclew/aclew10kq1/issues/1)

So I wonder what is the best conceptual solution? Should we routinely make a copy of `metadata/children.csv` and `metadata/recordings.csv` in our metrics repos?

**Comment by lucasgautheron** [comment, marked as answer]
I think it is a matter of choice and not something you can decide "in general".

So the only way to answer this is to answer the following questions first:
Are you willing to share the metadata? More generally, which parts of the data are you willing to share (with whom) ? What steps should be reproducible to anyone and what steps should be reproducible only for those who have access to the corpus?

Note that you do not need one subdataset for every step of the process (I would even recommend to avoid that). Nested subdatasets are especially desirable when each one may be useful by its own/have different accessibility settings.

In some cases, you could have one dataset for the paper, with the original corpus as the original subdatasets.
Then, in the paper dataset, you would have a number of scripts, one per step, where the first few steps would not be reproducible without the original data.

For instance, look at this dataset: https://gin.g-node.org/LAAC-LSCP/managing-storing-sharing-paper

There is a script for each figure that is generated from the data (https://gin.g-node.org/LAAC-LSCP/managing-storing-sharing-paper/src/main/code)

One of them requires the audio, the two others don't. In that case, it would be perfectly legitimate to release the paper as is, making it clear that only users that have access to the audio would be able to reproduce the figure that requires it.

*Now, to answer your case more specifically*: what I would do is that I would provide a script that extracts every information required for the rest of the analysis from the metadata into some intermediate file that is appropriate for public release, and use this file in the rest of the analysis.
This first processing step, of course, would not be reproducible by everyone. But the code being public, transparency -- and re-usability -- is still achieved in some way.
